util/abstract_fs.js

```diff
@@ -51,7 +51,13 @@ const _backends = {
         },
 
         async upload(localdir, url) {
-            await cmd.exec('aws', ['s3', 'sync', localdir, 's3://' + url.hostname + url.pathname]);
+            await this.uploadWithExtraArgs(localdir, url);
+        },
+
+        async uploadWithExtraArgs(localdir, url, ...extraArgs) {
+            const args = ['s3', 'sync', localdir, 's3://' + url.hostname + url.pathname];
+            if (extraArgs) args.push(...extraArgs);
+            await cmd.exec('aws', args);
         },
```

Comment:
Do we need two methods? `extraArgs` can be an optional parameter on the regular `upload` command.

Reply:
I was thinking of using one. Then I saw upload() has a special case of using `cp -rT`.  I want to preserve the old behavior.  If you're okay replacing it with rsync, I'll be happy to use one upload command.

Reply:
We can keep the special case only if extraArgs is empty, or just ditch it. I don't think it makes a huge difference, in practice we should be hitting the rsync path always, and rsync itself has a special case for local transfers.

Reply:
Done. I removed uploadWithExtraArgs and use rsync for local upload case as well.  It seems more consistent.